tests/helpers/datamodules.py

```diff
@@ -11,100 +11,51 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from typing import Any, Dict, Optional
+from typing import Optional
 
 import torch
-from sklearn.datasets import make_classification, make_regression
-from sklearn.model_selection import train_test_split
-from torch.utils.data import DataLoader, random_split
-from torch.utils.data.distributed import DistributedSampler
+from torch.utils.data import DataLoader
 
 from pytorch_lightning.core.datamodule import LightningDataModule
+from pytorch_lightning.utilities import _module_available
 from tests.helpers.datasets import MNIST, SklearnDataset, TrialMNIST
 
-
-class TrialMNISTDataModule(LightningDataModule):
-
-    def __init__(self, data_dir: str = "./"):
-        super().__init__()
-        self.data_dir = data_dir
-        self.non_picklable = None
-        self.checkpoint_state: Optional[str] = None
-
-    def prepare_data(self):
-        TrialMNIST(self.data_dir, train=True, download=True)
-        TrialMNIST(self.data_dir, train=False, download=True)
-
-    def setup(self, stage: Optional[str] = None):
-
-        if stage == "fit" or stage is None:
-            mnist_full = TrialMNIST(root=self.data_dir, train=True, num_samples=64, download=True)
-            self.mnist_train, self.mnist_val = random_split(mnist_full, [128, 64])
-            self.dims = self.mnist_train[0][0].shape
-
-        if stage == "test" or stage is None:
-            self.mnist_test = TrialMNIST(root=self.data_dir, train=False, num_samples=64, download=True)
-            self.dims = getattr(self, "dims", self.mnist_test[0][0].shape)
-
-        self.non_picklable = lambda x: x**2
-
-    def train_dataloader(self):
-        return DataLoader(self.mnist_train, batch_size=32)
-
-    def val_dataloader(self):
-        return DataLoader(self.mnist_val, batch_size=32)
-
-    def test_dataloader(self):
-        return DataLoader(self.mnist_test, batch_size=32)
-
-    def on_save_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
-        checkpoint[self.__class__.__name__] = self.__class__.__name__
-
-    def on_load_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
-        self.checkpoint_state = checkpoint.get(self.__class__.__name__)
+_SKLEARN_AVAILABLE = _module_available("sklearn")
```

Comment:
move this to `tests/__init__` as it could be useful elsewhere?

Reply:
Not a bad idea, but would rather do it in a separate PR